packages/@lwc/engine-core/src/framework/modules/static-parts.ts

```diff
@@ -105,3 +104,39 @@ export function applyStaticParts(
         applyRefs(part, owner);
     }
 }
+
+/**
+ * Reapplies static parts mounting to newly generated vnodes.
+ *
+ * @param n1 - the previous VStatic vnode
+ * @param n2 - the current VStatic vnode
+ * @param root - the root element
+ * @param renderer - the renderer to use
+ */
+export function patchStaticParts(n1: VStatic, n2: VStatic, root: Element, renderer: RendererAPI) {
+    if (!process.env.IS_BROWSER) {
+        return;
+    }
+
+    const { parts: currParts, owner: currPartsOwner } = n2;
+    if (isUndefined(currParts)) {
+        return;
+    }
+
+    const { parts: prevParts } = n1;
+    if (currParts.length !== prevParts?.length) {
+        // It should not be possible to reach this point as patching only occurs when the vnode is the same.
+        if (process.env.NODE_ENV !== 'production') {
+            throw new Error('Expected static parts to be the same for the same element');
```

Comment:
```suggestion
            throw new Error('Expected static parts to be the same for the same element. This is an error with the LWC framework itself.');
```

…Juuuuuust in case this ever actually makes it into production. 🤣

Reply:
There's no way that would happen! :stuck_out_tongue_winking_eye: